### PR TITLE
fix(data mapper): functions missing on load schema

### PR DESCRIPTION
**Pull request by DanielleCogs**
- **Please check if the PR fulfills these requirements**

* [x] The commit message follows our guidelines
* [x] Tests for the changes have been added (for bug fixes/features)
* [x] Docs have been added / updated (for bug fixes / features)

- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
bug fix
- **What is the current behavior?** (You can also link to an open issue here)
because of recent improvement, functions are not being shown on correct target nodes
- **What is the new behavior (if this is a feature change)?**
fixes the above
- **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
no
- **Please Include Screenshots or Videos of the intended change**:
N/A- app should be functioning as normal

### CODE REVIEW

File: __mocks__/mapDefinitions/FullTranscriptMapDefinition.ts [hunk 1]
```diff
@@ -1,144 +1,16 @@
-﻿export const fullTranscriptMapDefinitionString = `
```
Comment (refortie) [inline __mocks__/mapDefinitions/FullTranscriptMapDefinition.ts:1]: This change might be an accident?

File: libs/data-mapper/src/lib/models/Schema.ts [hunk 2]
```diff
@@ -64,7 +64,7 @@ export interface SchemaNodeExtended extends SchemaNode {
   pathToRoot: PathItem[];
   parentKey: string | undefined;
   // Used only in ArrayItem arrays, not regular arrays
-  arrayItemIndex: number | undefined;
+  arrayItemIndex?: number | undefined;
```
Comment (refortie) [inline libs/data-mapper/src/lib/models/Schema.ts:67]: Why make this optional?

File: libs/data-mapper/src/lib/utils/Function.Utils.ts [hunk 3]
```diff
@@ -134,3 +135,25 @@ export const functionsForLocation = (functions: FunctionDictionary, targetKey: s
   Object.fromEntries(
     Object.entries(functions).filter(([_key, value]) => value.functionLocations.some((location) => location.key === targetKey))
   );
+
+export const getFunctionLocationsForAllFunctions = (
+  dataMapConnections: ConnectionDictionary,
+  flattenedTargetSchema: SchemaNodeDictionary
+): FunctionDictionary => {
+  const functionNodes: FunctionDictionary = {};
```
Comment (refortie) [inline libs/data-mapper/src/lib/utils/Function.Utils.ts:143]: Nit: add a little white space to this to break it into logical sections?

**Review by refortie — CHANGES_REQUESTED** [review]
Remove the non-DM changes?

File: libs/data-mapper/src/lib/models/Schema.ts (hunk 2 above)
Comment (DanielleCogs) [reply libs/data-mapper/src/lib/models/Schema.ts:67]: it was giving me errors when I was adding the mocks that I pulled directly from the store. wouldn't it only apply to certain nodes anyways?

File: __mocks__/mapDefinitions/FullTranscriptMapDefinition.ts (hunk 1 above)
Comment (DanielleCogs) [reply __mocks__/mapDefinitions/FullTranscriptMapDefinition.ts]: nope I was using it for testing, fixed now

File: libs/data-mapper/src/lib/utils/Function.Utils.ts (hunk 3 above)
Comment (DanielleCogs) [reply libs/data-mapper/src/lib/utils/Function.Utils.ts:143]: done

**Review by refortie — APPROVED** [review]

**Pull request merged**